src/main/scala/org/constellation/trust/TrustManager.scala

```diff
@@ -46,14 +56,39 @@ class TrustManager[F[_]](nodeId: Id, clusterStorage: ClusterStorageAlgebra[F])(i
           trustNodes = TrustManager.calculateTrustNodes(trustNodesInternal.distinct, nodeId, scoringMap)
           //          _ <- logger.debug(s"TrustManager.calculateTrustNodes for trustNodes: ${trustNodes.toString()}")
 
-          idMappedScores: Map[Id, Double] = SelfAvoidingWalk
-            .runWalkFeedbackUpdateSingleNode(
-              selfNodeId,
-              trustNodes
-            )
-            .edges
-            .map(e => idxMap(e.dst) -> e.trust) //.flatMap(e => idxMap.get(e.dst).map(i => i -> e.trust) )//todo bug here
-            .toMap
+          sawAttempt = Try {
+            SelfAvoidingWalk
+              .runWalkFeedbackUpdateSingleNode(
+                selfNodeId,
+                trustNodes
+              )
+              .edges
+              .map(e => idxMap(e.dst) -> e.trust) //.flatMap(e => idxMap.get(e.dst).map(i => i -> e.trust) )//todo bug here
+              .toMap
+          }
+
+          // Temporary fallback
+          idMappedScores: Map[Id, Double] = sawAttempt match {
+            case Success(x) => x
+            case Failure(e) =>
+              logger.error(s"SAW Failure: $e")
```

Comment:
this logger will not work because it's wrapped in a monad

Reply:
Fixed